server/src/main/java/org/elasticsearch/common/settings/SettingsModule.java

```diff
@@ -175,13 +175,29 @@ private void registerSetting(Setting<?> setting) {
                 if (existingSetting != null) {
                     throw new IllegalArgumentException("Cannot register setting [" + setting.getKey() + "] twice");
                 }
+                if (setting.isConsistent()) {
+                    if (setting instanceof Setting.AffixSetting<?>) {
```

Comment:
I had not thought about consistent ordinary node settings at all. My reasoning is that we have a better way of actually enforcing consistency with cluster settings for all the non-secure node settings.
If we think we need a form of detection of inconsistency of other node settings, I would think of making use of the cluster scoped settings infra, to publish the actual values, rather than hashes as is the case here. In that case we might automatically achieve consistency, without admin intervention.
Can we leave this for another PR? I have updated the javadocs, there were just a few places that alluded to this aspect.